eventsourcing/src/main/java/org/axonframework/eventsourcing/EventCountSnapshotTriggerDefinition.java

```diff
@@ -76,24 +78,30 @@ public EventCountSnapshotTrigger(Snapshotter snapshotter, Class<?> aggregateType
 
         @Override
         public void eventHandled(EventMessage<?> msg) {
-            if (++counter >= threshold && msg instanceof DomainEventMessage) {
+            if (msg instanceof DomainEventMessage && ++counter >= threshold) {
                 if (CurrentUnitOfWork.isStarted()) {
-                    CurrentUnitOfWork.get().onPrepareCommit(
-                            u -> scheduleSnapshot((DomainEventMessage) msg));
+                    if (initialized) {
```

Comment:
Nit: wouldn't hurt to adjust the class level javadoc to specify both angles when a snapshot can be created.
Thus, firstly because during applying the threshold is reached, and secondly when the threshold is reached during initialization.